PolyEngine/Core/Src/String.cpp

```diff
@@ -69,12 +69,14 @@ String String::From(const std::string& var) {
 }
 
 bool String::Contains(const String& var) const {
-	Dynarray<size_t> idx = this->Data.FindAllIdx(*var.GetCStr());
-	// -1 because of null terminator
-	size_t dataSize = var.Data.GetSize() - 1;
-	for (size_t i = 0; i < idx.GetSize(); i++) {
-		String substr = this->Substring(idx[i], idx[i] + dataSize);
-		if (var == this->Substring(idx[i], idx[i] + dataSize)) {
+	size_t idx1 = 0, idx2 = 0;
+	for (idx1; idx1 < this->GetLength(); idx1++) {
+		if (var[idx2] == Data[idx1]) {
+			idx2++;
+		} else {
+			idx2 = 0;
+		}
+		if (idx2 == var.GetLength()) {
```

Comment:
We generally discourage usage of `this->`. You can also get rid of the brackets fo two if statements. Also each bracket should be put in separate line according to our coding style.

Reply:
@MuniuDev: I like how every statement in your comment is in direct opposition to my personal coding style, hahaha.